### DISCUSSION TITLE
Stock forecaster?

### DISCUSSION THREAD

**Discussion by StormTrooper** [inventree/InvenTree#5246] [Q&A]
Not sure if this is currently doable but is there a way to forecast when you will be running out of stock. 
Say I have 50 pieces of X, if I sell 10 a week and stock is tracked, then by  week 4 I would  like a alert  to say i will probably run out within a week and need to re-order.

I know you can set min stock but that's not the same as I might only use 1 piece every 6 months. 

Guess it's do able with a API and some fancy python scripts? If there is nothing built in to do this can anyone recommend a open source stock forecaster?

**Comment by matmair** [comment]
This is nothing that is integrated rn but should be possible and would be cool.
I can recommend https://frepple.com/ for complex forecasting/markets

**Comment by StormTrooper** [comment]
Thanks, this looks very impressive. Will take a stab at getting it installed and integrated into Inventree

**Comment by SchrodingersGat** [comment]
@StormTrooper we do have a [stock scheduling feature](https://docs.inventree.org/en/latest/part/scheduling/) which provides some of this functionality. If there are new features we could add or improve, let's discuss!

**Reply by StormTrooper** [reply]
Great, will look into this.  Was not aware of the scheduling feature.
